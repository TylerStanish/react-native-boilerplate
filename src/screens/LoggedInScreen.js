import React from 'react'
import {
  View,
  Text
} from 'react-native'
import {Button} from 'react-native-elements'
import {connect} from 'react-redux'

import Actions from '../actions'
import NavigationService from '../NavigationService'

class LoggedInScreen extends React.Component{

  componentWillReceiveProps(nextProps){
    if(this.props.loggedIn && !nextProps.loggedIn){
      NavigationService.navigate('SignedOut')
    }
  }

  render(){
    return(
      <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
        <Text>The user is now logged in</Text>
        <Button title='Click to logout' onPress={() => this.props.logout()}/>
      </View>
    )
  }
}

export default connect(state => {
  return {
    loggedIn: state.auth.loggedIn
  }
}, {logout: Actions.auth.logout})(LoggedInScreen)