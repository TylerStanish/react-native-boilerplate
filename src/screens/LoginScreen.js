import React from 'react'
import {
  View,
  Text
} from 'react-native'
import {
  Button
} from 'react-native-elements'
import {connect} from 'react-redux'

import Actions from '../actions'
import NavigationService from '../NavigationService'

class LoginScreen extends React.Component{

  componentWillReceiveProps(nextProps){
    if(!this.props.loggedIn && nextProps.loggedIn){
      NavigationService.navigate('SignedIn')
    }
  }

  render(){
    return(
      <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
        <Text>Hello from the login screen</Text>
        <Button title='Click to login' onPress={() => this.props.login()}/>
      </View>
    );
  }
}

export default connect(state => {
  return {
    loggedIn: state.auth.loggedIn
  }
}, {login: Actions.auth.login})(LoginScreen)