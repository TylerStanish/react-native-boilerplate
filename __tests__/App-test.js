import React from 'react'
import {
  Text
} from 'react-native'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {render} from 'react-native-testing-library'

import {LoggedInScreen} from '../src/screens/LoggedInScreen'
import {LoginScreen} from '../src/screens/LoginScreen'

const mockStore = configureStore([thunk])

describe('LoggedInScreen', () => {
  it('should match the last snapshot', () => {
    const props = {loggedIn: true}
    const tree = renderer.create(<LoggedInScreen {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('LoggedOutScreen', () => {
  it('should match the last snapshot', () => {
    const props = {loggedIn: false}
    const tree = renderer.create(<LoginScreen {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
