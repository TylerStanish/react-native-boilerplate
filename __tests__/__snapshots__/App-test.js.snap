// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInScreen should match the last snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    The user is now logged in 
    true
  </Text>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Click to logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`LoggedOutScreen should match the last snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello from the login screen
  </Text>
  <View
    style={
      Array [
        Object {
          "marginLeft": 15,
          "marginRight": 15,
        },
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        pointerEvents="box-only"
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#9E9E9E",
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 19,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "padding": 12,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontSize": 17.5,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          Click to login
        </Text>
      </View>
    </View>
  </View>
</View>
`;
